apollo-router/src/axum_http_server_factory.rs

```diff
@@ -539,6 +546,75 @@ fn prefers_html(accept_header: &HeaderValue) -> bool {
         .unwrap_or_default()
 }
 
+async fn decompress_request_body(
+    req: Request<Body>,
+    next: Next<Body>,
+) -> Result<Response, Response> {
+    let (parts, body) = req.into_parts();
+    let content_encoding = parts.headers.get(&CONTENT_ENCODING);
+    macro_rules! decode_body {
+        ($decoder: ident, $error_message: expr) => {{
+            let body_bytes = hyper::body::to_bytes(body)
+                .map_err(|err| {
+                    (
+                        StatusCode::BAD_REQUEST,
+                        format!("cannot read request body: {err}"),
+                    )
+                        .into_response()
+                })
+                .await?;
+            let mut decoder = $decoder::new(Vec::new());
+            decoder
+                .write_all(&body_bytes.to_vec())
```

Comment:
not dure you need to convert it to a vec first. It might even be possible to decompress in streaming